sympy/core/symbol.py

```diff
@@ -600,6 +600,53 @@ def matches(self, expr, repl_dict=None, old=False):
         return repl_dict
 
 
+class CodeblockResult(Symbol):
+    """
+    A result of a Codeblock.
+    """
+    _count = 0
+    _prng = random.Random()
+    _base_dummy_index = _prng.randint(10**6, 9*10**6)
+
+    __slots__ = ('dummy_index', 'expr', 'code_blocks')
+
+    is_Dummy = True
```

Comment:
Can't this just subclass from Dummy?

Reply:
Probably, will change it in the next commit.